### PR TITLE
Adding Feeds modules

### PR DESCRIPTION
**Pull request by sebastienros**
And integration with Contents and Lists (for blogs for instance)

### CODE REVIEW

**Comment by agriffard** [comment]
I enabled the Orchard.Feeds modules.

The url /Contents/Lists/{myBlogId}/rss displays the feed in Rss format.

I am wondering if I will be able to define an alias like this :  /feed.rss to point to this url  
Will there be an Alias admin page like in O1?

**Comment by agriffard** [comment]
In a separate enhancement, I would like to be able to add the meta <link rel="alternate" type="application/rss+xml" title="Feed title" href="http://myfeedurl" />.

It doesn't have to be necessarily automatic or in a Part setting because I would like to be able to specify the url of feed proxy. (FeedBurner for example)

If I could add anything in the head from a setting like I suggested here #422, it would fit my need.

**Comment by sebastienros** [comment]
Added Feed Proxy Url and Items count settings.

File: src/Orchard.Cms.Web/Modules/Orchard.Lists/Views/ListPart.Feed.cshtml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@inject IContentManager ContentManager
+
+@{ 
+    ContentItem contentItem = Model.ContentItem;
+    var contentItemMetadata = ContentManager.PopulateAspect<ContentItemMetadata>(contentItem);
+    var feedMetadata = ContentManager.PopulateAspect<FeedMetadata>(contentItem);
+    ResourceManager.RegisterLink(new LinkEntry
+    {
+        Rel = "alternate",
+        Type = "application/rss+xml",
```
Comment (agriffard) [inline src/Orchard.Cms.Web/Modules/Orchard.Lists/Views/ListPart.Feed.cshtml]: It is encoded when displayed in the source of the page:
application/rss&#x2B;xml

**Comment by agriffard** [comment]
I had to save the blog settings because FeedItemsCount was null.

Todo: Test that it works after a setup with a blog recipe and enabling the module.

Or will the feature be enabled by default?

**Comment by agriffard** [comment]
With this branch, I have a weird behavior: 
When I edit a blog post, the change is not taken in account when I modify any parts. If I go back to master, it works.

File: src/Orchard.ContentManagement.Abstractions/ContentElement.cs [hunk 2]
```diff
@@ -176,6 +176,13 @@ public void Weld(string name, object obj)
             {
                 Data[name] = JObject.FromObject(obj);
             }
+
+            // Also copy properties added to .Content
+            var contentPart = obj as ContentElement;
+            if (obj != null)
+            {
+                ((JObject)Data[name]).Merge(contentPart.Data, new JsonMergeSettings { MergeArrayHandling = MergeArrayHandling.Merge });
+            }
```
Comment (agriffard) [inline src/Orchard.ContentManagement.Abstractions/ContentElement.cs]: If I remove this code, I can edit and save.

Comment (sebastienros) [inline src/Orchard.ContentManagement.Abstractions/ContentElement.cs]: Thanks, taking a look. Too messy to deal between parts and dynamic data right now ...

**Comment by sebastienros** [comment]
Fixed the setup issues, and also refactored content element management quite a bit.

**Comment by agriffard** [comment]
Everything works for me.

**Comment by sebastienros** [comment]
The menu management has issues now, it was expected with the refactoring. Fixing it.

**Comment by sebastienros** [comment]
Ready to go

**Pull request merged**